### PR TITLE
feat: add Package.resolved swift files support

### LINKED ISSUE #4926: add support Swift Package Manager
## Description
Add support for finding vulnerabilities for `Swift Package Manager` (`Package.swift` files).
Use GHSA as datasource.

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add [Package.resolved](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#resolving-versions-packageresolved-file) files support.

Example of work:
```
➜  trivy fs Package.resolved
2023-08-23T15:44:40.465+0600    INFO    Vulnerability scanning is enabled
2023-08-23T15:44:40.465+0600    INFO    Secret scanning is enabled
2023-08-23T15:44:40.465+0600    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-08-23T15:44:40.465+0600    INFO    Please see also https://aquasecurity.github.io/trivy/dev/docs/scanner/secret/#recommendation for faster secret detection
2023-08-23T15:44:40.471+0600    INFO    Number of language-specific files: 1
2023-08-23T15:44:40.471+0600    INFO    Detecting swift vulnerabilities...

Package.resolved (swift)

Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

┌────────────────────────────┬───────────────┬──────────┬────────┬───────────────────┬────────────────────────┬────────────────────────────────────────────────────────┐
│          Library           │ Vulnerability │ Severity │ Status │ Installed Version │     Fixed Version      │                         Title                          │
├────────────────────────────┼───────────────┼──────────┼────────┼───────────────────┼────────────────────────┼────────────────────────────────────────────────────────┤
│ github.com/apple/swift-nio │ CVE-2022-3215 │ MEDIUM   │ fixed  │ 2.41.0            │ 2.29.1, 2.39.1, 2.42.0 │ SwiftNIO vulnerable to Improper Neutralization of CRLF │
│                            │               │          │        │                   │                        │ Sequences in HTTP Headers ('HTTP...                    │
│                            │               │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2022-3215              │
└────────────────────────────┴───────────────┴──────────┴────────┴───────────────────┴────────────────────────┴────────────────────────────────────────────────────────┘

```

## Related issues
- Close #4926

## Related PRs
- [x] aquasecurity/trivy-db/pull/339
- [x] aquasecurity/vuln-list-update/pull/233
- [x] aquasecurity/go-dep-parser/pull/245
- [x] aquasecurity/go-dep-parser/pull/250

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/language/swift/swift/swift.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	res, err := language.Analyze(types.Swift, input.FilePath, input.Content, p)
+	if err != nil {
+		return nil, xerrors.Errorf("%s parse error: %w", input.FilePath, err)
+	}
+	return res, nil
+}
+
+func (a swiftLockAnalyzer) Required(_ string, fileInfo os.FileInfo) bool {
+	return fileInfo.Name() == types.SwiftResolved
+}
```
Comment (knqyf263) [inline pkg/fanal/analyzer/language/swift/swift/swift.go:37]: I think other analyzers use `filePath`.

```suggestion
func (a swiftLockAnalyzer) Required(filePath string, _ os.FileInfo) bool {
	return path.Base(filePath) == types.SwiftResolved
}
```

**Review by knqyf263 — COMMENTED** [review]
We didn't add `ID` in `Package`. Didn't you see any implications?
https://github.com/aquasecurity/go-dep-parser/blob/e39b9645d9b842013a443b800af87608f010a532/pkg/swift/swift/parse.go#L37-L44

File: pkg/fanal/analyzer/language/swift/swift/swift.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/language/swift/swift/swift.go]: hm.. i don't know why i used FileInfo...
Thanks! Changed in https://github.com/aquasecurity/trivy/pull/4932/commits/9d1e8b6bc23008767689c224d72faf30669af9f3

**Comment by DmitriyLewen** [comment]
`Package.Resolved` files contain package names (e.g. https://github.com/aquasecurity/go-dep-parser/blob/e39b9645d9b842013a443b800af87608f010a532/pkg/swift/swift/testdata/happy-v1-Package.resolved#L5). 
But `Package.swift` files don't have names. To add dependencies you need to use URL in `Package.swift` file.

We usually use `<pkg_name>@<version>` schema for ID.
But use of package name can be confusing.

Peghaps we can use `<URL>@<version>` for `Package.Resolved`.

**Comment by knqyf263** [comment]
We already use `<URL>@<version>` schema for Go like `github.com/aquasecurity/trivy@v0.44.0`.

**Comment by DmitriyLewen** [comment]
You are right. I will add this.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**